indexer/osm_editor.cpp

```diff
@@ -0,0 +1,113 @@
+#include "osm_editor.hpp"
+
+#include "indexer/feature_decl.hpp"
+#include "indexer/index.hpp"
+
+#include "std/map.hpp"
+#include "std/set.hpp"
+
+namespace osm
+{
+// TODO(AlexZ, SergeyM): Use wrapper over XML (editor's "database" container).
+struct Editor::Impl
+{
+  // TODO(AlexZ): Use better structure(s) to store data, think about xml tree.
+  set<FeatureID> m_deletedFeatures;
+  map<FeatureID, FeatureType> m_editedFeatures;
+
+  Impl() { Load(); }
+  void Load()
+  {
+    // TODO(AlexZ): Initialize from xml file.
+  }
+
+  void Save()
+  {
+    // TODO(AlexZ): Save to xml file.
+  }
+};
+
+Editor::Editor() : m_impl(new Editor::Impl()) {}
+Editor::~Editor() { delete m_impl; }
+Editor::Impl & Editor::GetImpl()
+{
+  static Editor instance;
+  return *instance.m_impl;
+}
+
+bool Editor::IsFeatureDeleted(FeatureID const & fid)
+{
+  Impl & impl = GetImpl();
+  return impl.m_deletedFeatures.find(fid) != impl.m_deletedFeatures.end();
```

Comment:
Could you please implement a non-static Editor::IsFeatureDeletedImpl() method and use it here instead? It would be much simpler to check Editor for thread safety.